src/Mvc/Mvc.Core/src/DependencyInjection/ApiBehaviorOptionsSetup.cs

```diff
@@ -10,34 +10,38 @@
 
 namespace Microsoft.Extensions.DependencyInjection
 {
-    internal class ApiBehaviorOptionsSetup :
-        IConfigureOptions<ApiBehaviorOptions>,
-        IPostConfigureOptions<ApiBehaviorOptions>
+    internal class ApiBehaviorOptionsSetup : IConfigureOptions<ApiBehaviorOptions>
     {
-        internal static readonly Func<ActionContext, IActionResult> DefaultFactory = DefaultInvalidModelStateResponse;
-        internal static readonly Func<ActionContext, IActionResult> ProblemDetailsFactory =
-            ProblemDetailsInvalidModelStateResponse;
-
         public void Configure(ApiBehaviorOptions options)
         {
             if (options == null)
             {
                 throw new ArgumentNullException(nameof(options));
             }
 
-            options.InvalidModelStateResponseFactory = DefaultFactory;
+            options.InvalidModelStateResponseFactory = ProblemDetailsInvalidModelStateResponse;
             ConfigureClientErrorMapping(options);
-        }
 
-        public void PostConfigure(string name, ApiBehaviorOptions options)
-        {
-            // We want to use problem details factory only if
```

Comment:
This was done as a way to infer the compat setting from inside an option setup when migrating to 2.2. This cleans it up and removes a bunch of weird code that gets it right. If you're wondering, there's an actual setting on ApiBehaviorOptions to turn off the invalid model state filter that's still around.

Reply:
OK dank